src/api/cvc4cpp.cpp

```diff
@@ -2749,13 +2775,27 @@ std::vector<Sort> Solver::mkDatatypeSortsInternal(
   CVC4_API_SOLVER_TRY_CATCH_BEGIN;
 
   std::vector<CVC4::Datatype> datatypes;
-  for (size_t i = 0, ndts = dtypedecls.size(); i < ndts; i++)
-  {
-    CVC4_API_ARG_CHECK_EXPECTED(dtypedecls[i].getNumConstructors() > 0,
-                                dtypedecls[i])
+  for (size_t i = 0, ndts = dtypedecls.size(); i < ndts; ++i)
+  {
+    CVC4_API_ARG_AT_INDEX_CHECK_EXPECTED(this == dtypedecls[i].d_solver,
+                                         "datatype declaration",
+                                         dtypedecls[i],
+                                         i)
+        << "a datatype declaration associated to this solver object";
+    CVC4_API_CHECK(this == dtypedecls[i].d_solver)
+        << "Given datatype declaration at index " << i
+        << " is not associated with this solver";
```

Comment:
It looks like these two checks are the same? Am I missing something?

Reply:
Indeed. The second was replaced by the first but I forgot to delete it.